lib/scanner/walk_test.go

```diff
@@ -366,6 +366,43 @@ func TestWalkSymlinkWindows(t *testing.T) {
 	}
 }
 
+func TestWalkRootSymlink(t *testing.T) {
+
+	// Create a folder with a symlink to it
+
+	link := "link"
```

Comment:
Let's do this in a temp directory.

Reply:
Might be worth revisiting the TestWalkSymlink\Windows/Unix\]() functions in that case too.

Reply:
So don't take this as personal offence or something like that. It's more than likely that the existing tests code has issues. I am just pointing out things I see.

Reply:
No worries, not taken personally. :) The question should have asked is whether it would be helpful if I modified those tests while looking a this file? The reason I haven't is because it's out of scope of this pull request and I'm a bit uncomfortable modifying tests that are otherwise known to work correctly.

Reply:
I think do what you are doing here, and if you want to make other cleanups, file a separate PR.
This way if there is some disagreement or discussions around the refactoring, that doesn't get in the way of getting this merged.